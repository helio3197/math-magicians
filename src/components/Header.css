.menuButton {
  background: none;
  border: 0;
  font-size: 1.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.menuButton:hover {
  cursor: pointer;
}

@media (max-width: 494px) {
  .navLinksWrap {
    position: absolute;
    flex-direction: column;
    right: -24px;
    background: lightslategrey;
    top: 45px;
    padding: 12px 28px;
  }
}

@media (min-width: 495px) {
  .menuButton {
    display: none;
  }

  .navLinksWrap{
    display: flex !important;
    padding: 0;
  }
}