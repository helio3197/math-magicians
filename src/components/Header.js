import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Header.module.css';

const Header = () => {
  const navLinks = [
    {
      name: 'Home',
      path: '/',
      id: 1,
    },
    {
      name: 'Calculator',
      path: '/calculator',
      id: 2,
    },
    {
      name: 'Quote',
      path: '/quote',
      id: 3,
    },
  ];

  return (
    <div
      style={{
        width: '100%',
        display: 'flex',
        justifyContent: 'center',
        padding: '12px 24px',
      }}
    >
      <header
        style={{
          display: 'flex',
          width: '100%',
          maxWidth: '960px',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}
      >
        <h1
          style={{
            margin: '0',
          }}
        >
          Math Magicians
        </h1>
        <nav>
          <ul
            style={{
              margin: '0',
              padding: '0',
              listStyle: 'none',
              display: 'flex',
              gap: '16px',
            }}
          >
            {navLinks.map((item) => (
              <li className={styles.navItem} key={item.id}>
                <NavLink style={({ isActive }) => ((isActive) ? { color: 'red' } : {})} to={item.path}>{item.name}</NavLink>
              </li>
            ))}
          </ul>
        </nav>
      </header>
    </div>
  );
};

export default Header;
