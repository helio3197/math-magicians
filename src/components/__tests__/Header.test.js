import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header';

describe('Test for Header component', () => {
  test('The component should render', () => {
    const { asFragment } = render(
      <Router>
        <Header />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
