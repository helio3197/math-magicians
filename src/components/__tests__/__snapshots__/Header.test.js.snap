// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header component The component should render 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; display: flex; justify-content: center; padding: 12px 24px;"
  >
    <header
      style="display: flex; width: 100%; max-width: 960px; justify-content: space-between; align-items: center;"
    >
      <h1
        style="margin: 0px;"
      >
        Math Magicians
      </h1>
      <nav>
        <ul
          style="margin: 0px; padding: 0px; list-style: none; display: flex; gap: 16px;"
        >
          <li
            class="navItem"
          >
            <a
              aria-current="page"
              class="navActive"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              class=""
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              class=""
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</DocumentFragment>
`;
